### DISCUSSION TITLE
Recommended configuration for incorporating pgagroal with patroni

### DISCUSSION THREAD

**Discussion by darix** [pgagroal/pgagroal#193] [General]
Especially with regards to make the failover from one primary to a replica as seamless as possible.

**Comment by jesperpedersen** [comment]
There is no Patroni template for the project yet.

Any ideas of the best way to go on the subject ? The most common setup seems to be a HAProxy + pgbouncer kind of way which could be replicated.

**Comment by darix** [comment]
One thing i was wondering about:

you can publish DNS records with DCS systems like etcd+skydns/coredns or consul ... so if i have a DNS record with a very short TTL for the leader/primary, would pgagroal handle it gracefully with regards to caching retry when the primary node switches around due to failover/switchover.

**Reply by jesperpedersen** [reply]
Currently there is no TTL functionality in pgagroal. However, you can set a short `idle_timeout` value, and if you know the primary change you can issue a `flush-gracefully`.

The failover command is only run when pgagroal can't write to the current primary host, so I think in your case one or both options from above is better.

Any thoughts on improving TTL handling would be great.

**Comment by darix** [comment]
the failover handling I would probably leave to patroni. 

how long does pgagroal cache the dns result?

but having patroni trigger a "flush-gracefully" before the failover and hopefully wait for the "flush" to finish sounds like a neat way.

**Reply by jesperpedersen** [reply]
There is no caching of DNS entries at the moment. Any new connection will resolve its host name from the OS.
